modules/Users/controller.php

```diff
@@ -110,9 +111,86 @@ protected function action_saveuserwizard()
+        }
+        if (!empty($_POST['email1'])) {
+            $current_user->email1 = ($_POST['email1']);
+        }
+        if (!empty($_POST['phone_work'])) {
+            $current_user->phone_work = ($_POST['phone_work']);
+        }
+        if (!empty($_POST['phone_mobile'])) {
+            $current_user->phone_mobile = ($_POST['phone_mobile']);
+        }
+        if (!empty($_POST['messenger_type'])) {
+            $current_user->messenger_type = ($_POST['messenger_type']);
+        }
+        if (!empty($_POST['messenger_id'])) {
+            $current_user->messenger_id = ($_POST['messenger_id']);
+        }
+        if (!empty($_POST['address_street'])) {
+            $current_user->address_street = ($_POST['address_street']);
+        }
+        if (!empty($_POST['address_city'])) {
+            $current_user->address_city = ($_POST['address_city']);
+        }
+        if (!empty($_POST['address_state'])) {
+            $current_user->address_state = ($_POST['address_state']);
+        }
+        if (!empty($_POST['address_postalcode'])) {
+            $current_user->address_postalcode = ($_POST['address_postalcode']);
+        }
+        if (!empty($_POST['address_country'])) {
+            $current_user->address_country = ($_POST['address_country']);
+        }
+
+        // Saves User Details ONLY
+        $current_user->save();
+
+
+        // Will pull in the users Preferences from second page of the wizard
+        if (!empty($_POST['timezone'])){
+            $current_user->setPreference('timezone', $_POST['timezone'],
+                0, 'global');
+        }
+        if (!empty($_POST['dateformat'])) {
+            $current_user->setPreference('dateformat', $_POST['dateformat'],
+                0, 'global');
+        }
+        if (!empty($_POST['timeformat'])) {
+            $current_user->setPreference('timeformat', $_POST['timeformat'],
+                0, 'global');
+        }
+        if (!empty($_POST['currency'])) {
+            $current_user->setPreference('currency', $_POST['currency'],
+                0, 'global');
+        }
+        if (!empty($_POST['default_currency_significant_digits'])) {
+            $current_user->setPreference('default_currency_significant_digits',
+                $_POST['default_currency_significant_digits'], 0, 'global');
+        }
+        if (!empty($_POST['dec_sep'])) {
+            $current_user->setPreference('dec_sep', $_POST['dec_sep'],
+                0, 'global');
+        }
+        if (!empty($_POST['num_grp_sep'])) {
+            $current_user->setPreference('num_grp_sep', $_POST['num_grp_sep'],
+                0, 'global');
+        }
+        if (!empty($_POST['default_locale_name_format'])) {
+            $current_user->setPreference('default_locale_name_format',
+                $_POST['default_locale_name_format'], 0, 'global');
+        }
+
+        // saves User Preferences ONLY and redirect to new view
```

Comment:
Agreed with changing the comment here, although it should read something like:

```suggestion
        // Redirect to home
```

Since user prefs are already saved.